--- a/luci-app-mosdns/Makefile
+++ b/luci-app-mosdns/Makefile
@@ -6,7 +6,7 @@ PKG_RELEASE:=1
 
 LUCI_TITLE:=LuCI Support for mosdns
 LUCI_PKGARCH:=all
-LUCI_DEPENDS:=+mosdns-neo +jsonfilter +bash +luci-compat +v2ray-geoip +v2ray-geosite +zoneinfo-asia
+LUCI_DEPENDS:=+mosdns +jsonfilter +bash +luci-compat +v2ray-geoip +v2ray-geosite +zoneinfo-asia
 
 define Package/$(PKG_NAME)/conffiles
 /etc/mosdns/cus_config.yaml
