--- a/mosdns/Makefile
+++ b/mosdns/Makefile
@@ -44,11 +44,6 @@ define Package/mosdns
   DEPENDS:=$(GO_ARCH_DEPENDS)
 endef

-define Package/mosdns-neo
-  $(call Package/mosdns/Default)
-  DEPENDS:=$(GO_ARCH_DEPENDS)
-endef
-
 define Package/mosdns/config
 config MOSDNS_COMPRESS_GOPROXY
 	bool "Compiling with GOPROXY proxy"
@@ -81,10 +76,5 @@ define Package/mosdns/install
 	$(call GoPackage/Package/Install/Bin,$(1))
 endef

-define Package/mosdns-neo/install
-	$(call GoPackage/Package/Install/Bin,$(1))
-endef
-
 $(eval $(call GoBinPackage,mosdns))
 $(eval $(call BuildPackage,mosdns))
-$(eval $(call BuildPackage,mosdns-neo))
--