--- a/natflow/files/natflow-boot.init
+++ b/natflow/files/natflow-boot.init
@@ -14,11 +14,21 @@ disable_gro() {
 start() {
 	disable_gro
 	hwnat=$(uci get natflow.main.hwnat 2>/dev/null || echo 1)
+	[ "$(uci -q get network.lan.gateway)" ] && hwnat=0
 	enabled=$(uci get natflow.main.enabled 2>/dev/null || echo 1)
 	debug=$(uci get natflow.main.debug 2>/dev/null || echo 3)
 	delay_pkts=$(uci get natflow.main.delay_pkts 2>/dev/null || echo 0)
 	ifname_filter=$(uci get natflow.main.ifname_filter 2>/dev/null)
 
+	if [ "x${enabled}" = "x1" ]; then
+		if [ "x`uci get firewall.@defaults[0].flow_offloading 2>/dev/null`" = "x1" ]; then
+			uci set firewall.@defaults[0].flow_offloading=0
+			uci set firewall.@defaults[0].flow_offloading_hw=0
+			uci commit firewall
+			/etc/init.d/firewall reload
+		fi
+	fi
+
 	echo debug=$debug >/dev/natflow_ctl
 	echo disabled=$((!enabled)) >/dev/natflow_ctl
 	echo hwnat=$hwnat >/dev/natflow_ctl

--- a/natflow/files/natflow.config
+++ b/natflow/files/natflow.config
@@ -13,7 +13,7 @@ config auth
 	option enabled '1'
 	option szone '0'
 	option type 'auto'
-	option sipgrp '192.168.15.2-192.168.15.254'
+	option sipgrp '10.0.0.2-10.0.0.254'
 	option ipwhite ''
 	option macwhite ''
 
