--- a/pkg/conf/types.go
+++ b/pkg/conf/types.go
@@ -1,7 +1,5 @@
 package conf

-import "github.com/cloudreve/Cloudreve/v4/pkg/util"
-
 type DBType string

 const (
@@ -104,7 +102,7 @@
 // DatabaseConfig 数据库配置
 var DatabaseConfig = &Database{
 	Charset:     "utf8mb4",
-	DBFile:      util.DataPath("cloudreve.db"),
+	DBFile:      "/etc/cloudreve/cloudreve.db",
 	Port:        3306,
 	UnixSocket:  false,
 	DatabaseURL: "",
