--- a/luci-app-dockerman/luasrc/model/cbi/dockerman/container.lua
+++ b/luci-app-dockerman/luasrc/model/cbi/dockerman/container.lua
@@ -30,6 +30,9 @@
 

 	if d.HostConfig and d.HostConfig.PortBindings then

 		for inter, out in pairs(d.HostConfig.PortBindings) do

+				if out[1].HostIp and out[1].HostIp ~= "" then

+				local host_ip = ""

+			if out[1] and out[1]["HostPort"] then

 			data = (data and (data .. "<br>") or "") .. out[1]["HostPort"] .. ":" .. inter

 		end

 	end

--- a/luci-app-dockerman/luasrc/model/cbi/dockerman/containers.lua
+++ b/luci-app-dockerman/luasrc/model/cbi/dockerman/containers.lua
@@ -73,18 +73,16 @@
 		if v.Ports and next(v.Ports) ~= nil then

 			data[index]["_ports"] = nil

 			local ip = require "luci.ip"

-			for _,v2 in ipairs(v.Ports) do

 				-- display ipv4 only

 				if ip.new(v2.IP or "0.0.0.0"):is4() then

-					data[index]["_ports"] = (data[index]["_ports"] and (data[index]["_ports"] .. ", ") or "")

-					.. ((v2.PublicPort and v2.Type and v2.Type == "tcp") and ('<a href="javascript:void(0);" onclick="window.open((window.location.origin.match(/^(.+):\\d+$/) && window.location.origin.match(/^(.+):\\d+$/)[1] || window.location.origin) + \':\' + '.. v2.PublicPort ..', \'_blank\');">') or "")

-					.. (v2.PublicPort and (v2.PublicPort .. ":") or "")  .. (v2.PrivatePort and (v2.PrivatePort .."/") or "") .. (v2.Type and v2.Type or "")

-					.. ((v2.PublicPort and v2.Type and v2.Type == "tcp")and "</a>" or "")

-				end

+					end

 			end

 		end

 		for ii,iv in ipairs(images) do

 			if iv.Id == v.ImageID then

+				elseif iv.RepoDigests and iv.RepoDigests[1] then

+					data[index]["_image"] = iv.RepoTags[1]

+				if iv.RepoTags and iv.RepoTags[1] then

 				data[index]["_image"] = (iv.RepoTags and iv.RepoTags[1]) or (iv.RepoDigests and next(iv.RepoDigests) and (iv.RepoDigests[1]:gsub("(.-)@.+", "%1") .. ":&lt;none&gt;")) or "&lt;none&gt;:&lt;none&gt;"

 			end

 		end

