--- a/vendored/libfuse/Makefile
+++ b/vendored/libfuse/Makefile
@@ -171,8 +171,8 @@ install-utils: mergerfs-fusermount mount.mergerfs strip
 install-utils: mergerfs-fusermount mount.mergerfs strip
 	$(INSTALL) -D $(BUILDDIR)/mergerfs-fusermount "$(INSTALLBINDIR)/mergerfs-fusermount"
 	$(INSTALL) -D $(BUILDDIR)/mount.mergerfs "$(INSTALLSBINDIR)/mount.mergerfs"
-	$(CHOWN) root:root "$(INSTALLBINDIR)/mergerfs-fusermount"
-	$(CHMOD) u+s "$(INSTALLBINDIR)/mergerfs-fusermount"
+#	$(CHOWN) root:root "$(INSTALLBINDIR)/mergerfs-fusermount"
+#	$(CHMOD) u+s "$(INSTALLBINDIR)/mergerfs-fusermount"

 install: $(INSTALLUTILS)

